src/org/jetbrains/java/decompiler/modules/decompiler/SwitchHelper.java

```diff
@@ -313,14 +317,18 @@ private static boolean simplifyNewEnumSwitch(SwitchStatement switchSt) {
     if (inner instanceof InvocationExprent && ((InvocationExprent) inner).getName().equals("ordinal")) {
       InvocationExprent invInner = (InvocationExprent) inner;
 
-      ClassesProcessor.ClassNode classNode = DecompilerContext.getClassProcessor().getMapRootClasses().get(invInner.getClassname());
+      StructClass classStruct = DecompilerContext.getStructContext().getClass(invInner.getClassname());
+      if (classStruct == null) {
+        root.addComment("$VF: Unable to simplify switch-on-enum. You are probably missing external library", true);
```

Comment:
nitpick: Probably should be "... missing **an** external library"

Reply:
Fixed